### PR TITLE
fix(chart-heatfield): Chart HeatField is shown when start and end are…

### PR DESCRIPTION
**Pull request by sherifElhefnawyDynatrace**
### <strong>Pull Request</strong>

- Button/heatfield is now hidden if both start and end are undefined
- Added test to check for start and end being both undefined

#### Checklist
- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ffriedl89** [comment]
Could you maybe reword the commit to "fix(chart-heatfield): Heatfield is now hidden when start and end are undefined" - because this will end up in the changelog and might be confusing for people, because the fix does exactly the opposite of what you mention in the commit message ;)

**Comment by sherifElhefnawyDynatrace** [comment]
Reworded commit :)

**Comment by tomheller** [comment]
When the PR is merged, could the merging person please cherry pick the commits into the relevant branches (auto cherry picker does not work for forks)

File: libs/barista-components/chart/src/heatfield/chart-heatfield.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <button
   class="dt-chart-heatfield-marker"
   [class.dt-chart-heatfield-active]="active"
+  [style.visibility]="_isValidStartEndRange ? 'visible' : 'hidden'"
```
Comment (lukasholzer) [inline libs/barista-components/chart/src/heatfield/chart-heatfield.html]: Is it possible to click the invisible? Please check that :)
Maybe add an e2e test scenario for that

Comment (sherifElhefnawyDynatrace) [reply libs/barista-components/chart/src/heatfield/chart-heatfield.html]: Have already checked that, and here is a stackoverflow thread: https://stackoverflow.com/questions/5659563/css-is-a-hidden-object-clickable

Comment (sherifElhefnawyDynatrace) [reply libs/barista-components/chart/src/heatfield/chart-heatfield.html]: As for the e2e test, might add it later ya but not on this pr
